src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs

```diff
@@ -0,0 +1,626 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Microsoft.Win32.SafeHandles;
+using System.Collections.Generic;
+using System.Security;
+using System.Runtime.InteropServices;
+using System.Diagnostics.Contracts;
+using System.Linq;
+
+namespace System.IO.IsolatedStorage
+{
+    public sealed class IsolatedStorageFile : IDisposable
+    {
+        private string m_AppFilesPath;
+
+        private bool m_bDisposed;
+        private bool m_closed;
+
+        private object m_internalLock = new object();
```

Comment:
Ideally this would be ```readonly``` (I like that especially for locks, as accidentally changing the lock object could lead to very real and subtle problems).